grai-integrations/source-metabase/src/grai_source_metabase/loader.py

```diff
@@ -0,0 +1,295 @@
+        dbs_map (Dict[int, Dict]): A mapping of database IDs to their corresponding database dictionaries.
+        questions_map (Dict[int, Dict]): A mapping of question IDs to their corresponding question dictionaries.
+        namespace_map (Dict[int, str]): A mapping of database IDs to their corresponding namespace names.
+        question_table_map (Dict[int, int]): A mapping of question IDs to their corresponding table IDs.
+        table_db_map (Dict[int, int]): A mapping of table IDs to their corresponding database IDs.
+        question_db_map (Dict[int, int]): A mapping of question IDs to their corresponding database IDs.
+
+    """
+
+    def __init__(
+        self,
+        namespaces: Optional[Dict] = None,
+        default_namespace: Optional[str] = None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.default_namespace = default_namespace
+
+        # This line creates a list of tables by modifying each table dictionary obtained from the get_tables() method.
+        # It replaces the "schema" key with a new key "schema_name" while preserving the other key-value pairs.
+        # this is because the "schema" key is a reserved keyword in the pydantic.
+        self.tables = [
+            {**table, "schema_name": table.pop("schema")} for table in self.get_tables()
+        ]
+        self.tables_map = {table["id"]: table for table in self.tables}
+        self.dbs_map = {db["id"]: db for db in self.get_dbs()["data"]}
+        self.questions_map = {
+            question["id"]: question for question in self.get_questions()
+        }
+
+        self.namespace_map = build_namespace_map(
+            self.dbs_map, namespaces, default_namespace
+        )
+
+        self.question_table_map = {}
+        self.table_db_map = {}
+        self.question_db_map = {}
+
+    def build_lineage(self):
+        """
+        Builds the lineage information by updating the mappings and maps.
+
+        This method updates the `tables_map`, `dbs_map`, `questions_map`, `question_table_map`, `table_db_map`,
+        and `question_db_map` attributes.
+
+        """
+
+        self.question_table_map = {
+            question["id"]: question["table_id"]
+            for question in self.get_questions()
+            if question["table_id"] and question["archived"] is False
+        }
+
+        self.table_db_map = {table["id"]: table["db_id"] for table in self.tables}
+
+        for question, table in self.question_table_map.items():
+            self.question_db_map[question] = self.table_db_map[table]
+
+    def get_nodes(self) -> List[NodeTypes]:
+        """
+        Retrieves the list of nodes representing tables and questions.
+
+        Returns:
+            List[NodeTypes]: The list of nodes.
+
+        """
+
+        for question in self.questions_map.values():
+            question["namespace"] = self.namespace_map[
```

Comment:
The `Question` shouldn't be in the namespace of the database it refers to. There should be a namespace associated with metabase it lives inside.

Reply:
If the Question entities shouldn't be in the namespace of the db it refers to, what about the Table entities?

Reply:
So this is how I'm thinking about it:

Questions <-- query_from -- Tables
(Metabase)                              (A database or warehouse)
(Metabase namespace)          (The namespace of the database or warehouse)

We want to put things in the place they live so that everything connects up together and you know the Table in the database is, say, the same table being managed by dbt

Reply:
I have updated the pr with the responses.

Reply:
I think I can separate the questions into the ones with tables and the ones without tables that way we can set the namespace correctly